File: Algorithm/src/edu/Q2839.java
package edu;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Q2839 {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		int target = Integer.parseInt(br.readLine());
		int n = target/5;
		int remain = target%5;
		int c = 0;
		int result = 0; 
		
		
		while(c<=n) {
			remain = target - 5*(n-c);
			if(remain%3 == 0) {
				result = n - c + remain/3;
				break;
			}else {
				result = -1;
			}
			c = c + 1;
		}
		
		bw.write(Integer.toString(result));
		bw.close();
	}

}

File: to python/Q2839.py
import sys

if __name__ == "__main__":

    target = int(sys.stdin.readline())
    n = int(target/5)
    result = 0

    for i in range(n+1):
        remain = target - 5*(n-i)
        if remain%3 == 0:
            result = n - i + remain/3
            break
        else:
            result = -1

    print(int(result))
